### ISSUE TITLE
OLED Display header Pinout

### ISSUE THREAD

**Issue by marianol** [AndersBNielsen/65uino#2]
I was building my board and realized that the pinout of J3 (GND-VCC-SCL-SDA) did not matched the 0.96 Inch OLED Modules that I had, my modules have VCC-GND-SCL-SDA pinout. So the power input pins are other way around
This sent me on a search spree and it looks like this modules come in 2 flavors SSD1306 and SH1106. The SSD1306 has the pinout GND-VCC which works on the current PCB, while the SH1106 comes in the VCC-GND pinout which will require you to use a cable or adapter to switch the power leads. 

Will be a good addition to add some jumpers or 2 connectors in the PCB to accommodate for this. I know space is tight, once I get mine working I will give it a go.

**Comment by AndersBNielsen** [comment]
I have a feeling the SSD1306 and SH1106 aren't quite software compatible. They share most commands but I'm afraid it will take some code changes as well. 
SH1106 has four extra columns of RAM (132 vs 128 in the SSD1306) and a few extra commands as far as I can tell from a quick skim of the 1106 datasheet compared to my very volatile human memory. 

I'd love to hear your results - if it's close enough maybe I can add jumpers. If it's far off I should specify SSD1306 a bit better in the schematic.

**Comment by marianol** [comment]
Yes kept reading after I posted this and saw the memory size discrepancy between the two. 
I got a SSD1306 delivered today, solder the board, and burned the ROM all is working 100%.
I used a cable to connect the SH1106 that I already had and the welcome screen came right up. 
I will do some more testing next week, but it looks like although there are not exactly the same they do work. When I read the differences the framebuffer memory is off by one pixel, so the welcome message may be off by one and I did not notice. 
Will post here after more testing.

**Comment by AndersBNielsen** [comment]
The alternative pinout seems common enough that it makes sense to at least include solder jumpers with the current type preselected. (Starting work on the new revision "soon")
